File: 1509_Min_Diff_Between_Largest_And_Smallest_Value_In_Three_Moves.java
class Solution {
    public int minDifference(int[] nums) {
        if (nums.length < 5)
            return 0;
        Arrays.sort(nums);
        int result = Integer.MAX_VALUE;
        for (int i=0; i<4; i++) {
            result = Math.min(result, nums[nums.length-4+i] - nums[i]);
        }
        return result;
    }
}

File: Set1_1000/1509_Min_Diff_Between_Largest_And_Smallest_Value_In_Three_Moves.py
class Solution:
    def minDifference(self, nums: List[int]) -> int:
        n = len(nums)
        nums.sort()
        if n < 5:
            return 0
        return min(nums[n-1]-nums[3], nums[n-2]-nums[2], nums[n-3]-nums[1], nums[n-4]-nums[0])
